### ISSUE TITLE
twisted.internet.endpoints.serverFromString should be ported to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7982]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7982|
|Type|enhancement|
|Created|2015-08-01 04:03:09Z|
|Branch|https://github.com/twisted/twisted/tree/serverFromString-py-7982|

<details><summary>Searchable metadata</summary>

```
trac-id__7982 7982
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__serverFromString_py_7982 serverFromString-py-7982
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1438401789059805 1438401789059805
changetime__1465460582416196 1465460582416196
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45322]) Branching to serverFromString-py-7982.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

This is great. Thanks!

All looks good. Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45467]) Merge serverFromString-py-7982: Port twisted.internet.endpoints.serverFromString to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7982](https://github.com/twisted/twisted/issues/7982)
